### DISCUSSION TITLE
Ding state interval

### DISCUSSION THREAD

**Discussion by giorgiomanieri** [tsightler/ring-mqtt#673] [Feature Requests]
Is it possible to customize the duration of the Ding state? 
If it remains active for 20 seconds, it is impossible to activate the button combinations for opening the Ring Intercom. If the state could be autonomously modified (for example, to 1 second), it would be possible to set up an automation to open the gate if the doorbell is pressed 3 times within a short interval.

**Comment by tsightler** [comment]
As I don't own a Ring intercom, do you have some proof that Ring sends notifications for every button press even if they are only 1 second apart?  The code for Ring intercom is modeled after Ring doorbells because they appear to work basically the same way.  Ring doorbells certainly do not send a notification for every button press, actually, realistically only one notification every 30-60 seconds.

It's important to note that notifications are the only way ring-mqtt knows that a button has been pressed, the API does not provide real-time button state (actually, it doesn't provide any button state at all).  If you can show me a screenshot of the Ring app receiving multiple notifications in such a short period, I could consider such a feature.

> it is impossible to activate the button combinations for opening the Ring Intercom

Is this some Ring feature?

**Reply by giorgiomanieri** [reply]
Unfortunately, as you say, Ring doesn't provide more than one notification every 20 seconds, so the status remains on for about 20 seconds before being turned off.
It's probably a request to be forwarded to Ring support, as there could be many interesting features that could be added to the doorbell that are currently not possible.

**Reply by tsightler** [reply]
Realistically I think there is no way Ring would do notifications for each button press, however, they could implement local multi-press actions which produce different notification events based on long-press, double-press, triple-press, etc, which is the way most devices which support this kind of thing work.  That being said, I'd guess the chances of Ring ever doing this are nearly zero, they aren't exactly an innovative wonderland.  Good luck though!
